lib/Dialect/ModArith/IR/ModArithOps.td

```diff
@@ -15,63 +16,71 @@ class ModArith_Op<string mnemonic, list<Trait> traits = [Pure]> :
   let assemblyFormat = "operands attr-dict `:` functional-type(operands, results)";
 }
 
+def ModArith_ReduceOp : ModArith_Op<"reduce", [Pure, ElementwiseMappable]> {
+  let summary = "reduce a signed integer to its congruence modulo equivalent";
+
+  let description = [{
+  }];
+
+  let arguments = (ins
+    SignlessIntegerLike:$input
+  );
+  let results = (outs ModArithLike:$output);
+  let hasVerifier = 1;
+  let assemblyFormat = "operands attr-dict `:` type($input) `->` type($output)";
+}
+
+def ModArith_ExtractOp : ModArith_Op<"extract", [Pure, ElementwiseMappable]> {
+  let summary = "extract the integer value stored";
+
+  let description = [{
```

Comment:
Please add a description with an example syntax. I think the main goal of this op is to type convert, and it would ultimately be "free" (lower to zero machine instructions).